### PR TITLE
Add metrics for proxy and querynode

### LINKED ISSUE #17204: [Enhancement]: Add metrics for proxy and querynode
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to be added?

When refactoring querynode before, some metrics were lost, they are:

ProxyDecodeResultLatency  for query
ProxyReduceResultLatency  for query (always zero for now)

QueryNodeSQCount   // counter for search/query request arrived in querynode
QueryNodeSQReqLatency //  Time spent in search/query from querynode's  perspective
QueryNodeSQLatencyInQueue // duration which  search/query task  stayed in queue
QueryNodeReduceLatency    //Time spent in `reduce` step of search/query 

### Why is this needed?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by czs007**
fix #17204 
Signed-off-by: zhenshan.cao <zhenshan.cao@zilliz.com>

### CODE REVIEW

File: internal/querynode/impl.go [hunk 1]
```diff
@@ -697,16 +712,28 @@ func (node *QueryNode) Search(ctx context.Context, req *queryPb.SearchRequest) (
...
 func (node *QueryNode) Query(ctx context.Context, req *queryPb.QueryRequest) (*internalpb.RetrieveResults, error) {
+	metrics.QueryNodeSQCount.WithLabelValues(fmt.Sprint(Params.QueryNodeCfg.GetNodeID()), metrics.QueryLabel, metrics.TotalLabel).Inc()
 	failRet := &internalpb.RetrieveResults{
 		Status: &commonpb.Status{
 			ErrorCode: commonpb.ErrorCode_UnexpectedError,
 		},
 	}
+
+	defer func() {
+		if failRet.Status.ErrorCode != commonpb.ErrorCode_Success {
```
Comment (congqixia) [inline internal/querynode/impl.go:733]: `failRet.Status.ErrorCode` is always `UnexpectedError` in FromShardLeader branch

File: internal/querynode/impl.go [hunk 2]
```diff
@@ -560,11 +562,18 @@ func (node *QueryNode) isHealthy() bool {
...
 func (node *QueryNode) Search(ctx context.Context, req *queryPb.SearchRequest) (*internalpb.SearchResults, error) {
+	metrics.QueryNodeSQCount.WithLabelValues(fmt.Sprint(Params.QueryNodeCfg.GetNodeID()), metrics.SearchLabel, metrics.TotalLabel).Inc()
 	failRet := &internalpb.SearchResults{
 		Status: &commonpb.Status{
 			ErrorCode: commonpb.ErrorCode_UnexpectedError,
 		},
 	}
+
+	defer func() {
+		if failRet.Status.ErrorCode != commonpb.ErrorCode_Success {
```
Comment (congqixia) [inline internal/querynode/impl.go:573]: `failRet.Status.ErrorCode` is always `UnexpectedError` in FromShardLeader branch

**Comment by xiaofan-luan** [comment]
looks good to me~ just wait for ci pass

**Comment by czs007** [comment]
rerun ut

**Comment by czs007** [comment]
rerun ut

**Pull request merged**